src/field/arena.ts

```diff
@@ -301,8 +301,8 @@ export class Arena {
 
   /**
    * Attempts to set a new weather to the battle
-   * @param weather new weather to set of type WeatherType
-   * @param hasPokemonSource is the new weather from a pokemon
+   * @param weather {@linkcode WeatherType} new weather to set of type WeatherType
```

Comment:
```suggestion
   * @param weather {@linkcode WeatherType} new {@linkcode WeatherType} to set
```
you don’t need to put types into JS doc as we are using ts. This is just redundant